packages/hardhat-core/src/internal/hardhat-network/provider/blockchain/edr.ts

```diff
@@ -13,10 +13,12 @@ import {
 import { FilterParams } from "../node-types";
 import { bloomFilter, filterLogs } from "../filter";
 import { Bloom } from "../utils/bloom";
+import { EdrIrregularState } from "../EdrIrregularState";
 
 export class EdrBlockchain implements BlockchainAdapter {
   constructor(
     private readonly _blockchain: Blockchain,
+    private readonly _irregularState: EdrIrregularState,
```

Comment:
Maintain the irregular state for usage in retrieving a state at a particular block